ignite/metrics/metric.py

```diff
@@ -28,21 +31,21 @@ class MetricUsage:
             :meth:`~ignite.metrics.Metric.iteration_completed`.
     """
 
-    def __init__(self, started, completed, iteration_completed):
+    def __init__(self, started: Events, completed: Events, iteration_completed: CallableEventWithFilter) -> None:
```

Comment:
I'm a bit surprised with `iteration_completed: CallableEventWithFilter`, why it can't be `iteration_completed: Events`  ?

Reply:
It's actually because of this lines here
https://github.com/pytorch/ignite/blob/c8ad7dc0dbfc6b62251336b57fc504b648410ea7/ignite/metrics/metric.py#L117-L122
This usage suggests that the used parameter for `iteration_completed` is not of type `Events` but of `CallableEventWithFilter `.

Reply:
OK, I see, true that `Events.ITERATION_COMPLETED(*args, **kwargs)` returns an instance of `CallableEventWithFilter`:
https://github.com/pytorch/ignite/blob/11f6d6212cba56e6f7e20098a6e029421c7ac937/ignite/engine/events.py#L96

I wonder if this should not be replaced by something like 
```python
cast(Callable, type(self))(self.value, event_filter, self.name)
```
as `self` can be a derived class. On the other hand, maybe this does not worth a change...

Reply:
This results in an error... it looks like that the constructor of `Events` calls the `__call__` method of `Enum`, because `EventEnum(CallableEventWithFilter, Enum)` multi inherits. But somehow I don't get, why `Enum.__call__` is called and not `CallableEventWithFilter.__call__`?!
```
>       return type(self)(self.value, event_filter, self.name)
E       TypeError: __call__() takes from 2 to 3 positional arguments but 4 were given
```

Reply:
This is a bit strange. Let's keep it as you did.